deepspeed/profiling/flops_profiler/README.md

```diff
@@ -35,216 +38,324 @@ Top 3 modules in flops at depth 2 are {'Conv2d': '421.91 MMACs', 'Linear': '11.1
 Top 3 modules in params at depth 2 are {'Conv2d': '50.69 k', 'Linear': '11.01 k', 'Tanh': '0'}
 Top 3 modules in time at depth 2 are {'Conv2d': '12.25 ms', 'Linear': '6.88 ms', 'AvgPool2d': '3.23 ms'}
 Batch size:                     1024
-Number of multiply-adds:        439.55 MMACs
+Number of MACs:        439.55 MMACs
 Number of parameters:           61.71 k
-Number of steps profiled:       10
 ```
 
-## Installation
+## Supported Models
 
-The profiler is an integral part of DeepSpeed and can be installed by
+The flops estimation is partly inspired by [ptflops](https://github.com/sovrasov/flops-counter.pytorch) with the major difference being that the DeepSpeed flops profiler captures ```torch.nn.functional``` invoked in a module to estimate the flops. Thus the DeepSpeed flops profiler allows for customized modules in the model, e.g., ```ParallelTransformerLayerworks, ParallelSelfAttention, RowParallelLinear, etc.``` in [Megatron-LM](https://github.com/NVIDIA/Megatron-LM). This is in contrast to tools that profile at ```torch.nn.module``` level, such as ptflops, which require users to write customized flops calculation functions for each customized module. Finally, the DeepSpeed flops profiler also supports flops computation at module level (for RNNs).
 
-```
-pip install deepspeed
-```
+## Multi-GPU, Multi-node Runs
 
-Refer to the [installaiton of DeepSpeed](https://www.deepspeed.ai/getting-started/#installation) for more information.
+For models running on multi-GPU or multi-node, only the model parallelism (e.g. ```--model-parallel-size``` in [Megatron-LM](https://github.com/NVIDIA/Megatron-LM)) affects the number of flops and parameters profiled, i.e.,
+`model_parallel_size * flops = total_flops` and `model_parallel_size * parameters = total_parameters`. The number of GPUs or nodes does not affect the output profile.
 
-## Usage
 
-### With the DeepSpeed runtime
+## Usage With the DeepSpeed Runtime
```

Comment:
Can we have a Usage section with an opening paragraph that describes the following structure.? You already have the structure as below, I think having a paragraph that describes the structure upfront makes it easier for the reader to follow

Usage

With DeepSpeed Runtime

Without DeepSpeed Runtime

---Inference

---Training

Reply:
Made the changes as suggested. I added a table of content for the usage section as well.